src/commonMain/kotlin/fr/acinq/lightning/crypto/LocalKeyManager.kt

```diff
@@ -44,7 +44,11 @@ data class LocalKeyManager(val seed: ByteVector, val chain: Chain, val remoteSwa
     )
 
     override val finalOnChainWallet: KeyManager.Bip84OnChainKeys = KeyManager.Bip84OnChainKeys(chain, master, account = 0)
-    override val swapInOnChainWallet: KeyManager.SwapInOnChainKeys = KeyManager.SwapInOnChainKeys(chain, master, remoteSwapInServerKey)
+    override val swapInOnChainWallet: KeyManager.SwapInOnChainKeys = run {
+        val xpub = DeterministicWallet.ExtendedPublicKey.decode(remoteSwapInExtendedPublicKey).second
```

Comment:
We should check that the prefix returned by  `ExtendedPublicKey.decode` matches our chain

Reply:
Done in https://github.com/ACINQ/lightning-kmp/pull/482/commits/f5c599ba9fc7342a939de0110676e5d1828e7681